### DISCUSSION TITLE
Evaluate multilabel text classifier with different thresholds

### DISCUSSION THREAD

**Discussion by hp52** [explosion/spaCy#7095] [Help: Coding & Implementations] [answered]
Hi,
I trained a multilabel text classifier with Spacy 3.0 using the train command and the config file. The threshold was set to 0.5. The trained classifier is saved to <model_dir>. After training, I want to evaluate the text classifier on my test set.

`nlp = spacy.load(model_dir)`
`scorer = nlp.evaluate(test_data)`
`print(scorer["cats_macro_auc"])`
`print(scorer["cats_macro_p"])`
`print(scorer["cats_macro_r"])`
`print(scorer["cats_macro_f")`

This works fine and shows the ROC AUC score as well as the P, R, F scores for classification with threshold 0.5 (I guess that 0.5 is used because that was the threshold that was set in the config file). I'd like to explore the results with different thresholds. How can I specify a different threshold after training for my evaluation?

Thanks for your help.

**Comment by svlandeg** [comment, marked as answer]
You're right that the default threshold is set by the component, but it makes sense to want to change this dynamically when calculating scores.

https://github.com/explosion/spaCy/pull/6974#issuecomment-780614434 implements this functionality and will be available in the next release > 3.0.3. You'll be able to do:
```
nlp.evaluate(test_data, scorer_cfg={"threshold": 0.7})
```
and this new threshold will (temporarily) overwrite the threshold from the component itself.
